core/container/externalbuilder/session.go

```diff
@@ -24,13 +25,18 @@ type Session struct {
 	exited     chan struct{}
 	exitErr    error
 	waitStatus syscall.WaitStatus
+	signalChan <-chan struct{}
+	doneChan   chan<- struct{}
+	WaitGroup  sync.WaitGroup
 }
 
 // Start will start the provided command and return a Session that can be used
-// to await completion or signal the process.
+// to await completion or signal the process. The session will also listen for
+// any messages on the signalChan, at which time it will perform its cleanup
+// and send a message on the doneChan.
 //
 // The provided logger is used log stderr from the running process.
-func Start(logger *flogging.FabricLogger, cmd *exec.Cmd) (*Session, error) {
+func Start(logger *flogging.FabricLogger, cmd *exec.Cmd, signalChan <-chan struct{}, doneChan chan<- struct{}) (*Session, error) {
```

Comment:
I don't understand these two channels. The `signalChan` seems to be carrying a cancelation notification and I really don't understand the `doneChan`'s role given the `Wait` implementation.